src/dump.c

```diff
@@ -246,6 +247,28 @@ static void jl_load_sysimg_so(void)
         *sysimg_gvars[tls_offset_idx - 1] =
             (jl_value_t*)(uintptr_t)(jl_tls_offset == -1 ? 0 : jl_tls_offset);
 #endif
+        typedef void (*dispatch_t)(uint64_t, uint64_t, uint64_t, size_t*, void***, size_t**);
+        dispatch_t dispatchf = (dispatch_t)jl_dlsym(jl_sysimg_handle,
+                                                    "jl_dispatch_sysimg_fvars");
+        if (dispatchf) {
+            int32_t info[4];
+            jl_cpuid(info, 1);
+            int32_t infoex[4];
+            jl_cpuidex(infoex, 7, 0);
+            uint64_t mask = i32_to_i64(info[3], info[2]);
+            uint64_t emask1 = i32_to_i64(infoex[1], infoex[2]);
+            uint64_t emask2 = i32_to_i64(infoex[3], 0);
+            size_t nfunc = 0;
+            void **fptrs = NULL;
+            size_t *fidxs = NULL;
+            dispatchf(mask, emask1, emask2, &nfunc, &fptrs, &fidxs);
+            if (nfunc && fptrs && fidxs) {
+                for (size_t i = 0; i < nfunc; i++) {
+                    size_t fi = fidxs[i];
+                    sysimg_fvars[fi] = fptrs[i];
```

Comment:
I think I would rather leave this as a shared, read-only table and instead emit N tables (e.g. base table + (sparse?) overlay tables for each architecture). Otherwise we leak memory here, since we'll never want to reference it again after finishing loading. `dispatchf` seems still useful, to return the index of the appropriate overlay table to use (or, in the future, perhaps an ordered list of overlay tables).